src/gui/mainwindow.cpp

```diff
@@ -1,41 +1,39 @@
-#include <QtGui/QStandardItemModel>
-#include <QtWidgets/QHBoxLayout>
-#include <QtWidgets/QHeaderView>
-#include <QtWidgets/QMdiArea>
-#include <QtWidgets/QPushButton>
-#include <QtWidgets/QTableView>
-#include <QtWidgets/QToolBar>
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
 
 #include "candevice/candevice.h"
 #include "canrawsender/canrawsender.h"
 #include "canrawview/canrawview.h"
-#include "cansignalsender/cansignalsender.cpp"
+#include "cansignalsender/cansignalsender.h"
 #include "cansignalview/cansignalview.h"
 #include "mainwindow.h"
 
 MainWindow::MainWindow(QWidget* parent)
     : QMainWindow(parent)
+    , ui(new Ui::MainWindow)
     , mdi(std::make_unique<QMdiArea>())
     , canDevice(std::make_unique<CanDevice>(factory))
     , canRawView(std::make_unique<CanRawView>())
     , canSignalView(std::make_unique<CanSignalView>())
     , canRawSender(std::make_unique<CanRawSender>())
     , canSignalSender(std::make_unique<CanSignalSender>())
 {
-    mdi->setHorizontalScrollBarPolicy(Qt::ScrollBarAsNeeded);
-    mdi->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
-
-    setCentralWidget(mdi.get());
+    ui->setupUi(this);
+    ui->centralWidget->layout()->setContentsMargins(0,0,0,0);
 
     canRawView->setWindowTitle("Can Raw View");
-    mdi->addSubWindow(canRawView.get());
+    ui->mdiArea->addSubWindow(canRawView.get());
```

Comment:
unique_ptr is very bad here as Qt widget and windows takes over ownership, so in the end we have two "deletes". This is my bad, but can you correct it (and all remaining parts below) as you are changing related code? Furthermore these variables do not have to be member of MainWindow class.